src/monitor/nec_thread_monitor.cpp

```diff
@@ -45,10 +46,17 @@ NecThreadMonitor::NecThreadMonitor(Thread thread, trace::Trace& trace, NecDevice
 
 void NecThreadMonitor::monitor([[maybe_unused]] int fd)
 {
-    static int reg[] = { VE_USR_IC };
-    uint64_t val;
+    static int reg[] = {
+        VE_USR_PMC00, VE_USR_PMC01, VE_USR_PMC02, VE_USR_PMC03, VE_USR_PMC04, VE_USR_PMC05,
+        VE_USR_PMC06, VE_USR_PMC07, VE_USR_PMC08, VE_USR_PMC09, VE_USR_PMC10, VE_USR_PMC11,
+        VE_USR_PMC12, VE_USR_PMC13, VE_USR_PMC14, VE_USR_PMC15, VE_USR_IC,
+    };
 
-    auto ret = ve_get_regvals(device_.as_int(), nec_thread_.as_pid_t(), 1, reg, &val);
+    constexpr size_t num_counters = sizeof(reg) / sizeof(int);
+
+    uint64_t val[num_counters];
+
```

Comment:
```suggestion

    static_assert(reg[num_counters - 1] == VE_USR_IC, "last index must be instruction counter");

```

I think this should work as static_assert, if not, make it an `assert`